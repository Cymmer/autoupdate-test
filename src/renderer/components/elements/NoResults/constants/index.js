export { default as noResultsTypes } from './noResultsTypes';
