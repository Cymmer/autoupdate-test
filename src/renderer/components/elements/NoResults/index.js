import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import GLOBALS from 'codechum-app-globals';
import styles from './styles.module.scss';

import Text from '../Text';
import Cody from '../Cody';
import textTypes from '../Text/constants/textTypes';

import ButtonLink from '../Button/Link';
import Button from '../Button';

import noResultsTypes from './constants/noResultsTypes';

const NoResults = ({ id, text, className, codyType, action, type }) => (
  <div className={cn(styles[`NoResults___${type}`], className)}>
    <div className={styles.NoResults_imageContainer}>
      <Cody className={styles.NoResults_image} type={codyType} />
    </div>
    <Text id={id} type={textTypes.HEADING.XS}>
      {text}
    </Text>
    {action != null && (
      <>
        {typeof action.action === 'function' ? (
          <Button className={styles.NoResults_button} onClick={action.action}>
            {action.text}
          </Button>
        ) : (
          <ButtonLink className={styles.NoResults_button} to={action.action}>
            {action.text}
          </ButtonLink>
        )}
      </>
    )}
  </div>
);

NoResults.defaultProps = {
  id: 'noResultsFound',
  type: noResultsTypes.LARGE,
  text: 'No Results Found',
  codyType: GLOBALS.CODY_TYPES.ERROR,
  className: null,
  action: null,
};

NoResults.propTypes = {
  id: PropTypes.string,
  type: PropTypes.oneOf([noResultsTypes.LARGE, noResultsTypes.SMALL]),
  text: PropTypes.oneOfType(PropTypes.string, PropTypes.node),
  codyType: PropTypes.oneOf([
    GLOBALS.CODY_TYPES.WELCOME,
    GLOBALS.CODY_TYPES.ERROR,
    GLOBALS.CODY_TYPES.OK,
  ]),
  className: PropTypes.string,

  action: PropTypes.shape({
    action: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
    text: PropTypes.string,
  }),
};

export default NoResults;
