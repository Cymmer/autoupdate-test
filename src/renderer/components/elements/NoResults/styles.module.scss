@import '../../../scss/mixins';

.NoResults {
  &___large {
    @include flex-between;
    flex-direction: column;
    margin: 25vh 1rem;
    text-align: center;
  }

  &___small {
    @include flex-between;
    flex-direction: column;
    margin: 32px;
    text-align: center;
  }

  &_imageContainer {
    @include flex-center;
    margin: 0 0 16px 0;
  }

  &_image {
    height: 200px;
    width: 100%;
    min-width: 200px;
  }

  &_button {
    margin: 32px 0 0 0;
  }
}
