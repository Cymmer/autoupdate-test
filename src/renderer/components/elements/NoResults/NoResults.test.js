import React from 'react';
import { shallow } from 'enzyme';
import GLOBALS from 'codechum-app-globals';
import NoResults from './index';

describe('RENDERING - NoResults', () => {
  it('should render with default props', () => {
    const component = shallow(<NoResults />);
    expect(component).toMatchSnapshot();
  });

  it('should render the appropriate cody type', () => {
    let component = shallow(
      <NoResults codyType={GLOBALS.CODY_TYPES.WELCOME} />
    );
    expect(component).toMatchSnapshot();

    component = shallow(<NoResults codyType={GLOBALS.CODY_TYPES.ERROR} />);
    expect(component).toMatchSnapshot();
  });

  it('should render a button if it has a action of type function', () => {
    const component = shallow(
      <NoResults
        action={{
          action: () => {},
          text: 'Test',
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render a buttonlink if it has a action of type string', () => {
    const component = shallow(
      <NoResults
        action={{
          action: 'https://www.codechum.com',
          text: 'Test',
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
