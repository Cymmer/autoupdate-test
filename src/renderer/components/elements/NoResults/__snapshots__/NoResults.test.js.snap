// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - NoResults should render a button if it has a action of type function 1`] = `
<div
  className="NoResults___large"
>
  <div
    className="NoResults_imageContainer"
  >
    <Cody
      className="NoResults_image"
      src={null}
      type="error"
    />
  </div>
  <Text
    className={null}
    colorClass="n___700"
    element={null}
    id="noResultsFound"
    type="heading___xs"
  >
    No Results Found
  </Text>
  <Button
    className="NoResults_button"
    dataTestId={null}
    disabled={false}
    icon={null}
    id={null}
    isLoading={false}
    isLocked={false}
    kind="button"
    lockedIconClassName={null}
    onClick={[Function]}
    tabIndex={0}
    type="primary___blue"
    wrapperClassName={null}
  >
    Test
  </Button>
</div>
`;

exports[`RENDERING - NoResults should render a buttonlink if it has a action of type string 1`] = `
<div
  className="NoResults___large"
>
  <div
    className="NoResults_imageContainer"
  >
    <Cody
      className="NoResults_image"
      src={null}
      type="error"
    />
  </div>
  <Text
    className={null}
    colorClass="n___700"
    element={null}
    id="noResultsFound"
    type="heading___xs"
  >
    No Results Found
  </Text>
  <ButtonLink
    className="NoResults_button"
    disabled={false}
    icon={null}
    id={null}
    isLocked={false}
    onClick={null}
    tabIndex={0}
    to="https://www.codechum.com"
    type="primary___blue"
  >
    Test
  </ButtonLink>
</div>
`;

exports[`RENDERING - NoResults should render the appropriate cody type 1`] = `
<div
  className="NoResults___large"
>
  <div
    className="NoResults_imageContainer"
  >
    <Cody
      className="NoResults_image"
      src={null}
      type="welcome"
    />
  </div>
  <Text
    className={null}
    colorClass="n___700"
    element={null}
    id="noResultsFound"
    type="heading___xs"
  >
    No Results Found
  </Text>
</div>
`;

exports[`RENDERING - NoResults should render the appropriate cody type 2`] = `
<div
  className="NoResults___large"
>
  <div
    className="NoResults_imageContainer"
  >
    <Cody
      className="NoResults_image"
      src={null}
      type="error"
    />
  </div>
  <Text
    className={null}
    colorClass="n___700"
    element={null}
    id="noResultsFound"
    type="heading___xs"
  >
    No Results Found
  </Text>
</div>
`;

exports[`RENDERING - NoResults should render with default props 1`] = `
<div
  className="NoResults___large"
>
  <div
    className="NoResults_imageContainer"
  >
    <Cody
      className="NoResults_image"
      src={null}
      type="error"
    />
  </div>
  <Text
    className={null}
    colorClass="n___700"
    element={null}
    id="noResultsFound"
    type="heading___xs"
  >
    No Results Found
  </Text>
</div>
`;
